source/Irrlicht/COpenGLDriver.cpp

```diff
@@ -2236,27 +2236,23 @@ bool COpenGLDriver::SAuxContext::setActiveTexture(uint32_t stage, core::smart_re
 	return true;
 }
 
-void COpenGLDriver::SAuxContext::setActiveBindings(const COpenGLDescriptorSet* descriptorSets[4])
+void COpenGLDriver::SAuxContext::setActiveDescriptorSet(uint32_t index, const COpenGLDescriptorSet* ds)
 {
     //TODO optimize this
```

Comment:
First check if `ds` is actually different to previously bound (just check by pointer), then you can proceed.

then check for compatible layouts and the descriptor set preservation rules... unset higher indices if necessary. Handle a `null` that is empty descriptor set case.

Reply:
how does this play now with the state cache?

Reply:
@Crisspl status?

Reply:
resolved